internal/dns/srvRecordDialer_test.go

```diff
@@ -0,0 +1,184 @@
+package dns
+
+import (
+	// "net"
+	"context"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/foxcpp/go-mockdns"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewSRVRecordDialer(t *testing.T) {
```

Comment:
Consider using `github.com/stretchr/testify/suite` for tests. That package lets you organize your tests better, moving setup/teardown code into separate methods, etc. It provides a lot of the same structure as frameworks like TestNG for Java.